### PR TITLE
MORE INFO FOR SJ PREVIEWS: As Emmanuel, I want preview and harvest pages to be updated with ActionCable, so that I am able to see more information about what is going on with SJ

### PR DESCRIPTION
**Pull request by Reinkaos**
**Acceptance Criteria**

- Harvest operators can see timely information about progress and key events when running harvests, enrichments and previews
- UAT SJ stack used to minimise disruption to harvesters

### CODE REVIEW

File: app/assets/javascripts/channels/preview.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      // Called when the subscription is ready for use on the server
+    },
+
+    disconnected: function() {
+      // Called when the subscription has been terminated by the server
+    },
+
+    received: function(data) {
+
+      console.log(data);
```
Comment (EdwinRozario) [inline app/assets/javascripts/channels/preview.js]: remove

File: app/assets/javascripts/channels/preview.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      $('#harvest-attributes').html(data.harvested_attributes);
+      $('#api-record').html(data.api_record);
+
+      if(data.field_errors) {
+        $('#status').html("Field Errors");
+        $('#field-errors p').html(data.field_errors);
+        $('#field-errors').show();
+      }
+
+      if(data.deletable == true) {
```
Comment (tim-boost) [inline app/assets/javascripts/channels/preview.js]: Could use === here

File: app/assets/javascripts/channels/preview.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        $('#record-to-delete').show();
+      }
+
+      if(data.validation_errors) {
+        $('#status').html("Validation Errors");
```
Comment (tim-boost) [inline app/assets/javascripts/channels/preview.js]: Could use single quotes here

File: config/cable.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+#   adapter: async
+#
+# production:
+#   adapter: redis
+#   ur: redis-url
+#   channel_prefix: manager_production
+
+redis: &redis
+  adapter: redis
+  url: redis://localhost:6379/3
```
Comment (tim-boost) [inline config/cable.yml]: Would this need to be configurable for SJ Manager in the cloud?

Comment (Reinkaos) [reply config/cable.yml]: Yeah I'll make it an environment variable

**Pull request closed without merging**